### ISSUE TITLE
[Bug]: CKAN crashes during startup.

### ISSUE THREAD

**Issue by Syndel-Markus** [KSP-CKAN/CKAN#4247]
### Is there an existing issue for this?

- [X] I have checked the existing issues for an existing issue. I even actually looked at the ones that are pinned to the top!

### Operating System

Linux (Ubuntu 20.04)

### CKAN Version

1.35.2.24280

### Games

KSP 1

### Game Version

1.12.5

### Did you make any manual changes to your game folder (i.e., not via CKAN)?

No.

### Describe the bug

I had an older version installed. May have been 1.18 but not sure right now. Then I followed the steps on the [CKAN's APT repository](https://github.com/KSP-CKAN/CKAN/blob/master/doc/apt-repo.md) page to update my CKAN to a newer version. After update, when I clicked to start CKAN, I waited for it to load but nothing seemed to happen. So I opened a terminal window and typed "ckan" to see if there was something coming out. And that's how I discovered the crash. 

### Steps to reproduce

See above.

### Relevant log output

```
Unhandled Exception:
System.FormatException: Input string was not in a correct format.
  at System.Number.ThrowOverflowOrFormatException (System.Boolean overflow, System.String overflowResourceKey) [0x00020] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Number.ParseInt32 (System.ReadOnlySpan`1[T] value, System.Globalization.NumberStyles styles, System.Globalization.NumberFormatInfo info) [0x0001c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Int32.Parse (System.String s, System.Globalization.NumberStyles style, System.IFormatProvider provider) [0x00020] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Convert.ToInt32 (System.String value, System.IFormatProvider provider) [0x00008] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.String.System.IConvertible.ToInt32 (System.IFormatProvider provider) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) [0x0012d] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at ValveKeyValue.KVObjectValue`1[TObject].ToInt32 (System.IFormatProvider provider) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) [0x0012d] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType) [0x00007] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at ValveKeyValue.ObjectCopier.TryConvertValueTo[TValue] (System.Object value, TValue& converted) [0x00019] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00118] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
--- End of stack trace from previous location where exception was thrown ---

  at ValveKeyValue.ObjectCopier.InvokeGeneric (System.String methodName, System.Type genericType, System.Object[] parameters) [0x0007a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject (System.Type objectType, ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.CopyObject[TObject] (ValveKeyValue.KVObject kv, TObject obj, ValveKeyValue.IObjectReflector reflector) [0x000aa] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00108] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
--- End of stack trace from previous location where exception was thrown ---

  at ValveKeyValue.ObjectCopier.InvokeGeneric (System.String methodName, System.Type genericType, System.Object[] parameters) [0x0007a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject (System.Type objectType, ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.ConvertValue (System.Object value, System.Type valueType, ValveKeyValue.IObjectReflector reflector) [0x00016] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.ConvertValue[TValue] (System.Object value, ValveKeyValue.IObjectReflector reflector) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier+<>c__DisplayClass16_0`1[TElement].<MakeList>b__0 (System.Object i) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at System.Linq.Enumerable+SelectArrayIterator`2[TSource,TResult].ToList () [0x00017] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00021] in <d22af090bceb4be792f53595cf074724>:0 
  at ValveKeyValue.ObjectCopier.MakeList[TElement] (System.Object[] items, ValveKeyValue.IObjectReflector reflector) [0x0001f] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
--- End of stack trace from previous location where exception was thrown ---

  at ValveKeyValue.ObjectCopier.InvokeGeneric (System.String methodName, System.Type genericType, System.Object[] parameters) [0x0007a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier+<>c.<.cctor>b__26_0 (System.Type type, System.Object[] values, ValveKeyValue.IObjectReflector reflector) [0x0000b] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.ConstructTypedEnumerable (System.Type type, System.Object[] values, ValveKeyValue.IObjectReflector reflector, System.Object& typedEnumerable) [0x00061] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00079] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject) [0x00006] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.KVSerializer.Deserialize[TObject] (System.IO.Stream stream, ValveKeyValue.KVSerializerOptions options) [0x0001c] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.SteamLibrary.ShortcutsFileGames (ValveKeyValue.KVSerializer vdfParser, System.String path) [0x00019] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.SteamLibrary+<>c__DisplayClass1_0.<.ctor>b__4 (System.String p) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at System.Linq.Enumerable+SelectManySingleSelectorIterator`2[TSource,TResult].MoveNext () [0x0005f] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Collections.Generic.LargeArrayBuilder`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] items) [0x00046] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] items) [0x00000] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].ReserveOrAdd (System.Collections.Generic.IEnumerable`1[T] items) [0x00022] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Linq.Enumerable+Concat2Iterator`1[TSource].ToArray () [0x00016] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00021] in <d22af090bceb4be792f53595cf074724>:0 
  at CKAN.SteamLibrary..ctor (System.String libraryPath) [0x0012d] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.SteamLibrary..ctor () [0x0002a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.GameInstanceManager..ctor (CKAN.IUser user, CKAN.Configuration.IConfiguration configuration) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.GameInstanceManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0028d] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x000e9] in <e2499e8577f2402a87c97189d2e3067a>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.FormatException: Input string was not in a correct format.
  at System.Number.ThrowOverflowOrFormatException (System.Boolean overflow, System.String overflowResourceKey) [0x00020] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Number.ParseInt32 (System.ReadOnlySpan`1[T] value, System.Globalization.NumberStyles styles, System.Globalization.NumberFormatInfo info) [0x0001c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Int32.Parse (System.String s, System.Globalization.NumberStyles style, System.IFormatProvider provider) [0x00020] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Convert.ToInt32 (System.String value, System.IFormatProvider provider) [0x00008] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.String.System.IConvertible.ToInt32 (System.IFormatProvider provider) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) [0x0012d] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at ValveKeyValue.KVObjectValue`1[TObject].ToInt32 (System.IFormatProvider provider) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) [0x0012d] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType) [0x00007] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at ValveKeyValue.ObjectCopier.TryConvertValueTo[TValue] (System.Object value, TValue& converted) [0x00019] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00118] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
--- End of stack trace from previous location where exception was thrown ---

  at ValveKeyValue.ObjectCopier.InvokeGeneric (System.String methodName, System.Type genericType, System.Object[] parameters) [0x0007a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject (System.Type objectType, ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.CopyObject[TObject] (ValveKeyValue.KVObject kv, TObject obj, ValveKeyValue.IObjectReflector reflector) [0x000aa] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00108] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
--- End of stack trace from previous location where exception was thrown ---

  at ValveKeyValue.ObjectCopier.InvokeGeneric (System.String methodName, System.Type genericType, System.Object[] parameters) [0x0007a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject (System.Type objectType, ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.ConvertValue (System.Object value, System.Type valueType, ValveKeyValue.IObjectReflector reflector) [0x00016] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.ConvertValue[TValue] (System.Object value, ValveKeyValue.IObjectReflector reflector) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier+<>c__DisplayClass16_0`1[TElement].<MakeList>b__0 (System.Object i) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at System.Linq.Enumerable+SelectArrayIterator`2[TSource,TResult].ToList () [0x00017] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00021] in <d22af090bceb4be792f53595cf074724>:0 
  at ValveKeyValue.ObjectCopier.MakeList[TElement] (System.Object[] items, ValveKeyValue.IObjectReflector reflector) [0x0001f] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007c] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
--- End of stack trace from previous location where exception was thrown ---

  at ValveKeyValue.ObjectCopier.InvokeGeneric (System.String methodName, System.Type genericType, System.Object[] parameters) [0x0007a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier+<>c.<.cctor>b__26_0 (System.Type type, System.Object[] values, ValveKeyValue.IObjectReflector reflector) [0x0000b] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.ConstructTypedEnumerable (System.Type type, System.Object[] values, ValveKeyValue.IObjectReflector reflector, System.Object& typedEnumerable) [0x00061] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject, ValveKeyValue.IObjectReflector reflector) [0x00079] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.ObjectCopier.MakeObject[TObject] (ValveKeyValue.KVObject keyValueObject) [0x00006] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at ValveKeyValue.KVSerializer.Deserialize[TObject] (System.IO.Stream stream, ValveKeyValue.KVSerializerOptions options) [0x0001c] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.SteamLibrary.ShortcutsFileGames (ValveKeyValue.KVSerializer vdfParser, System.String path) [0x00019] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.SteamLibrary+<>c__DisplayClass1_0.<.ctor>b__4 (System.String p) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at System.Linq.Enumerable+SelectManySingleSelectorIterator`2[TSource,TResult].MoveNext () [0x0005f] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Collections.Generic.LargeArrayBuilder`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] items) [0x00046] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] items) [0x00000] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].ReserveOrAdd (System.Collections.Generic.IEnumerable`1[T] items) [0x00022] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Linq.Enumerable+Concat2Iterator`1[TSource].ToArray () [0x00016] in <d22af090bceb4be792f53595cf074724>:0 
  at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00021] in <d22af090bceb4be792f53595cf074724>:0 
  at CKAN.SteamLibrary..ctor (System.String libraryPath) [0x0012d] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.SteamLibrary..ctor () [0x0002a] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.GameInstanceManager..ctor (CKAN.IUser user, CKAN.Configuration.IConfiguration configuration) [0x00000] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.GameInstanceManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0028d] in <e2499e8577f2402a87c97189d2e3067a>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x000e9] in <e2499e8577f2402a87c97189d2e3067a>:0
```

**Comment by HebaruSan** [comment]
Hi @Syndel-Markus, thanks for the report. This was fixed in #4240 (specifically, c3f76f4365520306b965b0cb1579d89bb967bba2). You can download the dev build here:

- <https://ksp-ckan.s3-us-west-2.amazonaws.com/ckan.exe>

**Comment by Syndel-Markus** [comment]
> Hi @Syndel-Markus, thanks for the report. This was fixed in #4240 (specifically, [c3f76f4](https://github.com/KSP-CKAN/CKAN/commit/c3f76f4365520306b965b0cb1579d89bb967bba2)). You can download the dev build here:
> 
> * https://ksp-ckan.s3-us-west-2.amazonaws.com/ckan.exe

Thank you very much.
